container/dangerzone.py

```diff
@@ -45,10 +47,13 @@ async def read_stream(sr: asyncio.StreamReader, callback: Callable = None) -> by
         line = await sr.readline()
         if sr.at_eof():
             break
+        if (
+            os.environ.get("DZ_DEBUG_CONTAINER", "no") == "yes"
+            and line.decode().rstrip() != ""
```

Comment:
I wouldn't expect it to spit out binary. If it does so, I guess it's a bug we should investigate and having the decoding failure exception isn't too bad it seems. What do you think?

Reply:
Well, binary output in stderr is an indication of a bug, you're right. This may not be the case for stdout though. Also, there are other things that may go wrong, such as texts with different encoding. At the end of the day, if the stdout/stderr of an otherwise functioning program spits non-UTF-8 characters, we cannot abort the conversion. We can only abort it if the program fails.

So, my suggestion is to handle these characters differently instead. Python has some nice ways of doing that: https://docs.python.org/3/library/codecs.html#error-handlers. I'd propose going with the `'replace'` error handler.

Reply:
I saw the new version, but I'm still unsure about it. First, it still uses `line.decode()`, so it will crash if a non-UTF8 character is present. Secondly, now that I look at the code, I'm not even sure that it's needed in the first place. The `read_stream()` method returns the buffer it has read, so we can just grab it in the `run_command()` method, without having to check about the encoding, and adding more latency.

Reply:
After discussion we will not decode anything in the server component (container). If any extra processing needs to be done, we can do so in the client.

Reply:
This is now all done in the client-side.